.github/workflows/scripts/publish-sdkman.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+set -eu
+
+# latest stable dotty version 
+DOTTY_VERSION=$(curl -s https://api.github.com/repos/lampepfl/dotty/releases/latest  | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
+DOTTY_URL="https://github.com/lampepfl/dotty/releases/download/$DOTTY_VERSION/scala3-$DOTTY_VERSION.zip"
+
+# checking if dotty version is available 
+if ! curl --output /dev/null --silent --head --fail "$DOTTY_URL"; then
+  echo "URL not exists: $DOTTY_URL"
+  exit 1
+fi
+
+# Release a new Candidate Version
+curl -X POST \
+          -H "Consumer-Key: $SDKMAN_KEY" \
+          -H "Consumer-Token: $SDKMAN_TOKEN" \
+          -H "Content-Type: application/json" \
+          -H "Accept: application/json" \
+          -d '{"candidate": "scala", "version": "'"$DOTTY_VERSION"'", "url": "'"$DOTTY_URL"'"}' \
+          https://vendors.sdkman.io/release
```

Comment:
What happens if the POST fails?

I think what should happen is that we should stop the execution of the script here (and mark the workflow as a failed), and make sure a sensible error message is printed in the logs.

Reply:
I tested this case locally. For example,  I set invalid `SDKMAN_KEY` and `SDKMAN_TOKEN`.
After running script, I received the following error:
`curl: (22) The requested URL returned error: 403 Forbidden`

and stop execution of the script with 22 exit code.